packages/scripts/scripts/plugin-zip.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+const AdmZip = require( 'adm-zip' );
+const { sync: glob } = require( 'fast-glob' );
+const { sync: packlist } = require( 'npm-packlist' );
+const { dirname } = require( 'path' );
+const { stdout } = require( 'process' );
+
+/**
+ * Internal dependencies
+ */
+const { hasPackageProp, getPackageProp } = require( '../utils' );
+
+const name = getPackageProp( 'name' );
+stdout.write( `Creating archive for \`${ name }\` plugin... 🎁\n\n` );
+const zip = new AdmZip();
+
+let files = [];
+if ( hasPackageProp( 'files' ) ) {
```

Comment:
Just an idea here, but could we have the `files` entry merge with the existing defaults? This would make it much easier to add a new directory/file without needing to also provide all of the defaults.

Reply:
So that exists already? That would be great to add to the docs in the PR.

Reply:
Not here, but it would be simple to replicate.

Reply:
> Just an idea here, but could we have the files entry merge with the existing defaults? This would make it much easier to add a new directory/file without needing to also provide all of the defaults.

Let's keep it as is for now. It's a bit tricky to handle with `files` because it is used also when publishing to npm. The good news is that when you set `files` then you don't need to list files like readme, changelog, or license because they are always included. I figured out that we can't add any custom items in `files` because it would break `npm publish` if someone is using it.